examples/ChatQnA/deploy/k8s_helm.md

```diff
@@ -0,0 +1,497 @@
+environment variable or `values.yaml` 
+
+Set a new [namespace](#create-and-set-namespace) and switch to it if needed
+
+To enable UI, uncomment the lines `56-62` in `GenAIInfra/helm-charts/chatqna/values.yaml`:
+```bash
+chatqna-ui:
+   image:
+     repository: "opea/chatqna-ui"
+     tag: "latest"
+   containerPort: "5173"
+```
+
+Next, we will update the dependencies for all Helm charts in the specified directory and ensure the `chatqna` Helm chart is ready for deployment by updating its dependencies as defined in the `Chart.yaml` file.
+
+```bash
+# all Helm charts in the specified directory have their 
+# dependencies up-to-date, facilitating consistent deployments.
+./update_dependency.sh
+
+# "chatqna" here refers to the directory name that contains the Helm 
+# chart for the ChatQnA application 
+helm dependency update chatqna
+```
+
+Set the necessary environment variables to setup the use case
+```bash
+export MODELDIR="/mnt/opea-models"  #export MODELDIR="null" if you don't want to cache the model.  
+export MODELNAME="Intel/neural-chat-7b-v3-3"
+export EMBEDDING_MODELNAME="BAAI/bge-base-en-v1.5"
+export RERANKER_MODELNAME="BAAI/bge-reranker-base"
+```
+
+## Deploy the use case
+The `helm install` command will initiate all the aforementioned services such as Kubernetes pods.
+
+```bash
+helm install chatqna chatqna \
+    --set global.HUGGINGFACEHUB_API_TOKEN=${HF_TOKEN} \
+    --set global.modelUseHostPath=${MODELDIR} \
+    --set tgi.LLM_MODEL_ID=${MODELNAME} \
+    --set tei.EMBEDDING_MODEL_ID=${EMBEDDING_MODELNAME} \
+    --set teirerank.RERANK_MODEL_ID=${RERANKER_MODELNAME}
+```
+
+**OUTPUT:**
+```bash
+NAME: chatqna
+LAST DEPLOYED: Thu Sep  5 13:40:20 2024
+NAMESPACE: chatqa
+STATUS: deployed
+REVISION: 1
+```
+It takes a few minutes for all the microservices to be up and running. Go to the next section which is [Validate Microservices](#validate-microservices) to verify that the deployment is successful.
+
+
+### Validate microservice
+#### Check the pod status
+To check if all the pods have started, run:
+
+```bash
+kubectl get pods
+``` 
+You should expect a similar output as below:
+```
+NAME                                      READY   STATUS    RESTARTS      AGE
+chatqna-chatqna-ui-77dbdfc949-6dtms        1/1     Running   0          5m7s
+chatqna-data-prep-798f59f447-4frqt         1/1     Running   0          5m7s
+chatqna-df57cc766-t6lkg                    1/1     Running   0          5m7s
+chatqna-nginx-5dd47bfc7d-54x96             1/1     Running   0          5m7s
+chatqna-redis-vector-db-7f489b6bb6-mvzbw   1/1     Running   0          5m7s
+chatqna-retriever-usvc-6695979d67-z5jgx    1/1     Running   0          5m7s
+chatqna-tei-769dc796c-gh5vx                1/1     Running   0          5m7s
+chatqna-teirerank-54f58c596c-76xqz         1/1     Running   0          5m7s
+chatqna-tgi-7b5556d46d-pnzph               1/1     Running   0          5m7s
```

Comment:
Could not get all pods to run, filed https://github.com/opea-project/GenAIExamples/issues/1202

Reply:
@arun-gupta I was thinking of adding separate resource requirements for - on-prem (baremeral) servers and cloud instances
Does that make sense?